src/docfx/build/page/BuildPage.cs

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class BuildPage
+    {
+        private static readonly Type[] s_schemaTypes = new[] { typeof(LandingData) };
+        private static readonly IReadOnlyDictionary<string, Type> s_schemas = s_schemaTypes.ToDictionary(type => type.Name);
+
+        public static Task<(IEnumerable<Error> errors, PageModel result, DependencyMap dependencies)> Build(
+            Document file,
+            TableOfContentsMap tocMap,
+            ContributionInfo contribution,
+            Action<Document> buildChild)
+        {
+            Debug.Assert(file.ContentType == ContentType.Page);
+
+            var dependencies = new DependencyMapBuilder();
+
+            var (errors, pageType, content, htmlTitle, wordCount, fileMetadata) = Load(file, dependencies, buildChild);
+
+            var locale = file.Docset.Config.Locale;
+            var metadata = JsonUtility.Merge(Metadata.GetFromConfig(file), fileMetadata);
+            var (id, versionIndependentId) = file.Docset.Redirections.TryGetDocumentId(file, out var docId) ? docId : file.Id;
+
+            // TODO: add check before to avoid case failure
+            var (repoErrors, author, contributors, updatedAt) = contribution.GetContributorInfo(
+                file,
+                metadata.Value<string>("author"),
+                metadata.Value<DateTime?>("update_date"));
+
+            var (editUrl, contentUrl, commitUrl) = contribution.GetGitUrls(file);
```

Comment:
>contribution.GetGitUrls(file); [](start = 51, length = 30)

the logic to generate these metadata is different between conceptual and sdp. See [https://ceapex.visualstudio.com/Engineering/_workitems/edit/29187](https://ceapex.visualstudio.com/Engineering/_workitems/edit/29187) There will be more diversity like this, e.g. Markdown fragement. Still keep md and sdp separate?

Reply:
for legacy format they are different, in v3 we could consolidate them